File: ProblemCollection/src/kakao_blind_recruitment2019/FailureRate.js
function solution(N, stages) {
    const answer = Array(N)
        .fill(1)
        .map((v, i) => v + i);
    const arr = Array(N + 1).fill(0);
    for (let i of stages) arr[i - 1]++;
    let count = arr[N];
    for (let i = N - 1; i >= 0; i--) {
        count += arr[i];
        arr[i] /= count;
    }
    return answer.sort((a, b) => arr[b - 1] - arr[a - 1]);
}

File: ProblemCollection/src/kakao_blind_recruitment2019/FailureRate.py
def solution(N, stages):
    answer = [i for i in range(1, N + 1)]
    arr = [0] * (N + 1)
    for i in stages:
        arr[i - 1] += 1
    count = arr[N]
    for i in range(N - 1, -1, -1):
        count += arr[i]
        if count == 0:
            continue
        arr[i] /= count
    return sorted(answer, key=lambda x: arr[x - 1], reverse=True)
